egs/wsj/s5/steps/nnet3/lstm/make_configs.py

```diff
@@ -191,25 +196,30 @@ def ParseLstmDelayString(lstm_delay):
     prev_layer_output = nodes.AddLdaLayer(config_lines, "L0", prev_layer_output, args.config_dir + '/lda.mat')
 
     for i in range(args.num_lstm_layers):
-	if len(lstm_delay[i]) == 2: # BLSTM layer case, add both forward and backward
+        if len(lstm_delay[i]) == 2: # BLSTM layer case, add both forward and backward
+            if args.preserve_state == 'true':
+                raise ValueError("State preserving training with BLSTM is not desirable.")
             prev_layer_output1 = nodes.AddLstmLayer(config_lines, "BLstm{0}_forward".format(i+1), prev_layer_output, args.cell_dim,
                                              args.recurrent_projection_dim, args.non_recurrent_projection_dim,
+                                             args.preserve_state,
                                              args.clipping_threshold, args.norm_based_clipping,
                                              args.ng_per_element_scale_options, args.ng_affine_options,
                                              lstm_delay = lstm_delay[i][0])
             prev_layer_output2 = nodes.AddLstmLayer(config_lines, "BLstm{0}_backward".format(i+1), prev_layer_output, args.cell_dim,
                                              args.recurrent_projection_dim, args.non_recurrent_projection_dim,
+                                             args.preserve_state,
                                              args.clipping_threshold, args.norm_based_clipping,
                                              args.ng_per_element_scale_options, args.ng_affine_options,
                                              lstm_delay = lstm_delay[i][1])
             prev_layer_output['descriptor'] = 'Append({0}, {1})'.format(prev_layer_output1['descriptor'], prev_layer_output2['descriptor'])
```

Comment:
With a new commit from @danpovey it might no longer be necessary to ensure the correct ordering of descriptors as the c++ code checks and corrects it. Please check if you can eliminate the descriptor specific changes you made to support BLSTMs.